tests/unit/test_validate_metadata.py

```diff
@@ -8,54 +11,80 @@
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 
-INPUT_DIR = 'tests/resources/input_directory'
-VALID_METADATA_FILE_PATHS = [
-    f'{INPUT_DIR}/SYNT_BEFOLKNING_SIVSTAND/SYNT_BEFOLKNING_SIVSTAND.json',
-    f'{INPUT_DIR}/SYNT_PERSON_INNTEKT/SYNT_PERSON_INNTEKT.json'
-]
-VALID_METADATA_REF_PATH = (
-    f'{INPUT_DIR}/SYNT_BEFOLKNING_KJOENN/SYNT_BEFOLKNING_KJOENN.json'
-)
-NO_SUCH_METADATA_FILE = 'NO/SUCH/FILE'
-MISSING_IDENTIFIER_METADATA_FILE_PATH = (
-    f'{INPUT_DIR}/MISSING_IDENTIFIER_DATASET/MISSING_IDENTIFIER_DATASET.json'
-)
-INVALID_SENSITIVITY_METADATA_FILE_PATH = (
-    f'{INPUT_DIR}/INVALID_SENSITIVITY_DATASET/INVALID_SENSITIVITY_DATASET.json'
-)
-EMPTY_CODELIST_METADATA_FILE_PATH = (
-    f'{INPUT_DIR}/EMPTY_CODELIST_DATASET/EMPTY_CODELIST_DATASET.json'
-)
+INPUT_DIRECTORY = 'tests/resources/input_directory'
+WORKING_DIRECTORY = 'tests/resources/working_directory'
+EXPECTED_DIRECTORY = 'tests/resources/expected/validate_metadata'
+
+VALID_METADATA = ['SYNT_BEFOLKNING_SIVSTAND', 'SYNT_PERSON_INNTEKT']
+VALID_METADATA_REF = 'SYNT_BEFOLKNING_KJOENN'
+
+NO_SUCH_METADATA = 'NO_SUCH_METADATA'
+MISSING_IDENTIFIER_METADATA = 'MISSING_IDENTIFIER_DATASET'
+INVALID_SENSITIVITY_METADATA = 'INVALID_SENSITIVITY_DATASET'
+EMPTY_CODELIST_METADATA = 'EMPTY_CODELIST_DATASET'
 REF_DIRECTORY = 'tests/resources/ref_directory'
 
 
 def test_validate_valid_metadata():
-    for metadata_file_path in VALID_METADATA_FILE_PATHS:
+    for metadata in VALID_METADATA:
         data_errors = validate_metadata(
-            metadata_file_path
+            metadata,
+            input_directory=INPUT_DIRECTORY,
+            working_directory=WORKING_DIRECTORY,
+            keep_temporary_files=True
         )
+        with open(
+            Path(WORKING_DIRECTORY) / f'{metadata}.json',
+            'r',
+            encoding='utf-8'
+        ) as f:
+            actual_metadata = json.load(f)
+        with open(
+            Path(EXPECTED_DIRECTORY) / f'{metadata}.json',
+            'r',
+            encoding='utf-8'
+        ) as f:
+            expected_metadata = json.load(f)
+        assert actual_metadata == expected_metadata
```

Comment:
Unless I'm missing something: two files (actual_metadata and expected_metadata) are read and compared, no custom logic here that is being tested 🤔 The other assert is OK though.

Reply:
The actual_metadata is the metadata that is produced by the validate_metadata function, while the expected_metadata is what the validate_metadata is expected to produce based on the input data. So its similar to:

```py
expected_amount = 8
actual_amount = count_people_in_room(ROOM_WITH_8_PEOPLE)
assert actual_amount == expected_amount
```

Reply:
Right, validate_metadata creates a file that you read in the test 👍

Reply:
That being said: We might need to update these tests in the future, they might be too statically defined and stringent. If we update a descriptive field in the central unit type definitions they will fail f.ex. Let's keep an eye on them 👀